tests/test_s3.py

```diff
@@ -1003,6 +1003,35 @@ def complete(self):
         eq_(True, MockMultipartS3Upload.aborted)
         eq_("Error!", rep.mirror_exception)
 
+    @parameterized.expand([
+        ('default_expiration_parameter', None, int(S3Uploader.S3_DEFAULT_PRESIGNED_URL_EXPIRATION)),
+        ('empty_expiration_parameter', {S3Uploader.S3_PRESIGNED_URL_EXPIRATION: 100}, 100)
+    ])
+    def test_sign_url(self, name, expiration_settings, expected_expiration):
+        # Arrange
+        region = 'us-east-1'
+        bucket = 'bucket'
+        filename = 'filename'
+        url = 'https://{0}.s3.{1}.amazonaws.com/{2}'.format(bucket, region, filename)
+        expected_url = url + '?AWSAccessKeyId=KEY&Expires=1&Signature=S'
```

Comment:
Minor, but instead of testing that the query params are present and this order, could you instead test that they (and no others) are present and have the correct values?

Reply:
Another way of doing this would be to guarantee a deterministic order by sorting the keys when generating the URL. These keys are sorted but I think this code is in a library we don't control, so I don't know if that's guaranteed.

Reply:
This test uses `expected_url` as a return value of the mocked `generate_presigned_url` method:
```python
s3_uploader.client.generate_presigned_url = MagicMock(return_value=expected_url)
```
It's done to not use the real `boto3` in the test and this guarantees that `sign_url` will return exactly this value